### PR TITLE
:sparkles: Add profiler support to front proxy and virtual workspaces

### PR DESCRIPTION
**Pull request by bipuladh**
## Summary
Adds profiler address as command line argument

Fixes https://github.com/kcp-dev/kcp/issues/1836

Screenshot: 
Screenshot from 2022-09-19 20-44-37

### CODE REVIEW

File: pkg/virtual/options/options.go [hunk 1]
```diff
@@ -72,6 +76,11 @@ func (o *Options) NewVirtualWorkspaces(
 	wildcardKubeInformers kubernetesinformers.SharedInformerFactory,
 	wildcardKcpInformers kcpinformers.SharedInformerFactory,
 ) ([]rootapiserver.NamedVirtualWorkspace, error) {
+
```
Comment (stevekuznetsov) [inline pkg/virtual/options/options.go:79]: I think the main `kcp` process also calls this - if so, we'd want to make sure that we start the profiler only when we're in the standalone case.

Comment (bipuladh) [reply pkg/virtual/options/options.go:79]: @stevekuznetsov I don't think we would be able to set the profiler address when it's getting called from `kcp` process but I could move the invocation to `cmd.go`.

Comment (stevekuznetsov) [reply pkg/virtual/options/options.go:79]: When we're running the virtual workspace server as part of `kcp`, we should be using the `kcp` options to run the (single) profiler server. When we're running standalone, we should use the built-in k8s APIServer profiling option for the VW to run it.

Comment (bipuladh) [reply pkg/virtual/options/options.go:78]: I am not sure how to use the profiling option of APIServer and specify a particular address as it's config doesn't accept any endpoints as per  https://pkg.go.dev/k8s.io/apiserver/pkg/server?utm_source=gopls#Config.EnableProfiling
@sttts any pointers?

Comment (sttts) [reply pkg/virtual/options/options.go:78]: yes, this belongs into cmd/virtual-workspaces/options.

**Comment by s-urbaniak** [comment]
just a linter check error for the missing err check after the ListenAndServe call, else lgtm from my side as well 👍

File: cmd/virtual-workspaces/command/cmd.go [hunk 2]
```diff
@@ -96,7 +98,7 @@ func Run(ctx context.Context, o *options.Options) error {
 	identityConfig, resolveIdentities := bootstrap.NewConfigWithWildcardIdentities(nonIdentityConfig, bootstrap.KcpRootGroupExportNames, bootstrap.KcpRootGroupResourceExportNames, nil)
 	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Millisecond*500, func(ctx context.Context) (bool, error) {
 		if err := resolveIdentities(ctx); err != nil {
-			klog.V(3).Infof("failed to resolve identities, keeping trying: %v", err)
+			logger.V(3).Info(fmt.Sprintf("failed to resolve identities, keeping trying: %v", err))
```
Comment (sttts) [inline cmd/virtual-workspaces/command/cmd.go:101]: this is not yet what we need here (although the linter passed): look into other uses of logger throughout the code-base. Basically if you need `fmt.Sprintf` it's a sign of non-structured logs. Structured logs have pairs of keys and values behind the string. Here `err` is such a value. @stevekuznetsov is the expert here. Maybe he has some good pointers.

File: cmd/virtual-workspaces/command/cmd.go [hunk 3]
```diff
@@ -161,7 +168,7 @@ func Run(ctx context.Context, o *options.Options) error {
 		return err
 	}
 
-	klog.Infof("Starting virtual workspace apiserver on %s (%s)", rootAPIServerConfig.GenericConfig.ExternalAddress, version.Get().String())
+	logger.Info(fmt.Sprintf("Starting virtual workspace apiserver on %s (%s)", rootAPIServerConfig.GenericConfig.ExternalAddress, version.Get().String()))
```
Comment (sttts) [inline cmd/virtual-workspaces/command/cmd.go:171]: similarly to above

**Pull request merged**